tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/Tcl.xtext

```diff
@@ -51,11 +51,23 @@ TestCleanup:
 	contexts+=TestStepContext*
 ;
 
-SpecificationStepImplementation:
+SpecificationStepImplementation hidden(SPACES, SL_COMMENT, ML_COMMENT):
 	{SpecificationStepImplementation}
-	'*' contents+=StepContent* '.'?
+	'*' contents+=tslStepContent* nl=NL
 	contexts+=TestStepContext*;
 
+tslStepContent returns tsl::StepContent: 
+	tslStepContentText | StepContentVariable;
+
+tslStepContentText returns tsl::StepContentText:
+	value=Word;
+
+Word returns ecore::EString hidden():
+	ID | StepContentTextOperator | INT | '*' | '@' | ',' | '|' | '.' | '?' | '!' | '#' | '##' | '${' | '%' | '&' | '(' | ')' | '-' | '=' | 
```

Comment:
what if you grouped these in parentheses, e.g. `( … '*' | '@' | ',' | …)+`
Wouldn't that force the scanner to consume whole "words" made up of these letters and symbols, instead of producing separate tokens? Or does this cause other issues? Am I barking up the wrong tree?

Reply:
yeah, why not. currently this is not used anywhere but I agree that connected characters may form a token and should thus be modeled accordingly.

Reply:
I have to leave it this way, since the parser will report ambiguities that cannot be resolved. Since this is not really an issue right now, I'll leave it at that